### ISSUE TITLE
[NOSTR] Add a Zap button in the chat window that allows the user to zap a contact with a preset amount

### ISSUE THREAD

**Issue by islandbitcoin** [lnflash/flash-mobile#27]
- [ ] Add a zap icon in the chat window that allows a user to zap the contact they are chatting with. The zap button should deduct the bitcoin from the users default wallet. If the contact is a flash user without a nostr profile set up, the default amount should be sent to the users flash lightning address.
- [ ] Add a field to the settings screen, allowing the user to set the default amount of sats that will be zapped to the contact.

**Comment by ecurrencyhodler** [comment]
Bounty [posted](https://community.sphinx.chat/tickets?owner_id=021ae436bcd40ca21396e59be8cdb5a707ceacdb35c1d2c5f23be7584cab29c40b&created=1696811901)!

**Comment by KoalaSat** [comment]
Hi! I can take a look to this, just a question, is this default amount the only way to set a zap's monto?

**Comment by islandbitcoin** [comment]
Hey! 

The requirement is flexible, but the default amount was inspired by this screen in damus:
![Screenshot 2023-12-22 at 8 25 50 AM](https://github.com/lnflash/flash-mobile/assets/34528298/59872714-a39f-4a2d-8eb0-616d3c81078f)

If you think this flow could be easier for the user to zap, I'm open to suggestions.

**Comment by KoalaSat** [comment]
I see, maybe we can just implement these exact same choices and allow the user to set a custom amount on an input bellow?

**Comment by islandbitcoin** [comment]
> I see, maybe we can just implement these exact same choices and allow the user to set a custom amount on an input bellow?

that would be ideal

**Comment by KoalaSat** [comment]
Alright, if you are not into hurries, I can work on this after Christmas 😃

**Comment by islandbitcoin** [comment]
I'm aiming to create an MVP of the app to demo on January 9th
I can hide the nostr chat tab for the demo if it isn't ready by then

**Comment by ecurrencyhodler** [comment]
Welcome @KoalaSat if you're interested, we have a bitcoin bounty posted for this. Let me know if you're interested in claiming it after you address this issue. I'll help you get onboarded to Sphinx and can assign it to you. 

https://community.sphinx.chat/bounty/512

Also do you mind me asking how you came across this project?

**Comment by KoalaSat** [comment]
> Welcome @KoalaSat if you're interested, we have a bitcoin bounty posted for this. Let me know if you're interested in claiming it after you address this issue. I'll help you get onboarded to Sphinx and can assign it to you.
> 
> https://community.sphinx.chat/bounty/512
> 
> Also do you mind me asking how you came across this project?

Nice I'll take a look, it was @islandbitcoin the one who contacted me via nostr to take a look to the project 😃

**Comment by KoalaSat** [comment]
Ey @ecurrencyhodler I was trying to login into Sphinx, isn't there any option to just signin with no installing their stuff?

**Comment by ecurrencyhodler** [comment]
@KoalaSat you could try LNURL but you'd have to handle payment outside the platform!

**Comment by ecurrencyhodler** [comment]
I can also give you a free sphinx code if you want!

**Comment by KoalaSat** [comment]
> I can also give you a free sphinx code if you want!

It would be helpful thanks! I just don't want to pay or install extra software on my node

**Comment by ecurrencyhodler** [comment]
Okay great! 

You don't need to install it on your node. You can use our hosted version on mobile or desktop. Get it here: https://buy.sphinx.chat/

And here's the unique sphinx code to sign up as a new user (check email):
connect::relay.swarm4.sphinx.chat::0224f1243261da0516143d689c13c8eb895ee0c372f80a082e0836c8c75020abf9

**Comment by Nightrider0098** [comment]
@KoalaSat are you still working on this issue?

**Comment by KoalaSat** [comment]
@Nightrider0098 nope sorry, I couldn't find a way to claim it without installing the software or paying

**Comment by Nightrider0098** [comment]
@islandbitcoin is this issue still open?

**Comment by islandbitcoin** [comment]
no, sorry, I've closed the bounty since we are working on the nostr solution using a more holistic approach, but you are very welcome to contribute! Here is the current active PR being worked on

https://github.com/lnflash/flash-mobile/pull/102

**Comment by turizspace** [comment]
is this bounty open to be claimed? i saw it on #sphinx

**Comment by islandbitcoin** [comment]
No, we closed bounties and moved to just open source contributions.

**Comment by turizspace** [comment]
are the contributions compensated? or does this mean you want volunteer code?

**Comment by islandbitcoin** [comment]
> are the contributions compensated? or does this mean you want volunteer code?

Yes. 
https://opensource.guide/how-to-contribute/#why-contribute-to-open-source

**Comment by islandbitcoin** [comment]
Contract work is also possible, send me a DM on nostr

**Comment by turizspace** [comment]
> Contract work is also possible, send me a DM on nostr

sent a DM. thank you so much :)

**Comment by abh3po** [comment]
[]  - Send a Zap request on zaps
[]  - Default zap amounts & one tap default
[]  - Custom Zap amount on hold

**Comment by turizspace** [comment]
> is this bounty open to be claimed? i saw it on #sphinx

aah... the good old days of bounties. when next contract job?
